test/integration.test.js

```diff
@@ -1035,6 +1021,39 @@ describe('lint-staged', () => {
     expect(await execGit(['rev-list', '--count', 'HEAD'])).toEqual('2')
     expect(await readFile('--looks-like-flag.js')).toEqual(testJsFilePretty)
   })
+
+  /**
+   * Ugly hacks to skip the following tests on Windows, until escaping filenames is figured out
+   */
+  const itSkipWindows = process.platform === 'win32' ? it.skip : it
+
+  itSkipWindows('should handle simple filename with shell option', async () => {
+    const FILENAME = 'test.js'
+    await appendFile(FILENAME, testJsFileUgly)
+    await execGit(['add', '--', FILENAME])
+    await expect(gitCommit({ ...fixJsConfig, shell: true })).resolves.toEqual(undefined)
+    expect(await execGit(['rev-list', '--count', 'HEAD'])).toEqual('2')
+    expect(await readFile(FILENAME)).toEqual(testJsFilePretty)
+  })
+
+  itSkipWindows('should handle complex filename with shell option', async () => {
+    const FILENAME = `&& touch 'evil.js'' && test.js`
```

Comment:
Shouldn't matter, as the test simply checks if `touch 'evil.js'` would run, creating the file. If `evil.js` doesn't exist, the filename was properly escaped. I'll remove the extra `'` because that would probably break the shell preventing the command to run.